internal/api/tests/achievement_handler_test.go

```diff
@@ -60,125 +98,143 @@ func TestCreateAchievementHandler(t *testing.T) {
 				}`,
 			setup:              func(mockSvc *mocks.Service) {},
 			expectedStatusCode: http.StatusBadRequest,
+			expectedResponse:   `{"error_code":400,"error_message":"parameter missing : name "}`,
 		},
 		{
-			name: "Fail_for_missing_description_field",
+			name: "Fail_when_CreateAchievement_fails",
 			input: `{
 				"achievements":[{
-				    "name": "Star Performer",
-					"description": ""
-				    }]
-				}`,
-			setup:              func(mockSvc *mocks.Service) {},
-			expectedStatusCode: http.StatusBadRequest,
+					"name": "Star Performer",
+					"description": "Description of Award"
+				}]
+			}`,
+			setup: func(mockSvc *mocks.Service) {
+				mockSvc.On("CreateAchievement", mock.Anything, mock.AnythingOfType("specs.CreateAchievementRequest"), 1, 1).Return(0, errors.New("service failure")).Once()
+			},
+			expectedStatusCode: http.StatusBadGateway,
+			expectedResponse:   `{"error_code":502,"error_message":"service failure"}`,
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			test.setup(profileSvc)
-
-			req, err := http.NewRequest("POST", "/profiles/achievements", bytes.NewBuffer([]byte(test.input)))
-			if err != nil {
-				t.Fatal(err)
-				return
-			}
+			defer profileSvc.AssertExpectations(t)
+			req := httptest.NewRequest("POST", "/profiles/achievements", bytes.NewBuffer([]byte(test.input)))
 			req = mux.SetURLVars(req, map[string]string{"profile_id": "1"})
 			rr := httptest.NewRecorder()
 
-			ctx := context.WithValue(req.Context(), userIDKey, 1)
+			ctx := context.WithValue(req.Context(), constants.UserIDKey, 1.0)
 			req = req.WithContext(ctx)
 
 			handler := http.HandlerFunc(createAchievementHandler)
 			handler.ServeHTTP(rr, req)
 
 			if rr.Result().StatusCode != test.expectedStatusCode {
-				t.Errorf("Expected %d but got %d", test.expectedStatusCode, rr.Result().StatusCode)
+				t.Errorf("Expected status %d but got %d", test.expectedStatusCode, rr.Result().StatusCode)
+			}
+
+			if rr.Body.String() != test.expectedResponse {
+				t.Errorf("Expected response body %s but got %s", test.expectedResponse, rr.Body.String())
 			}
 		})
 	}
 }
 
-// Define a custom type for context key
-type contextKey string
-
-// Define constants for context keys
-const (
-	userIDKey        contextKey = "user_id"
-	profileIDKey     contextKey = "profile_id"
-	achievementIDKey contextKey = "achievement_id"
-)
-
 func TestUpdateAchievementHandler(t *testing.T) {
 	achSvc := new(mocks.Service)
-
+	updateAchievementHandler := handler.UpdateAchievementHandler(context.Background(), achSvc)
 	tests := []struct {
 		name               string
 		input              string
 		setup              func(mockSvc *mocks.Service)
 		expectedStatusCode int
+		expectedResponse   string
 	}{
 		{
 			name: "Success_for_achievement_update",
```

Comment:
Add test case for
- Fail GetMultipleParams
- Fail GetUserIDFromContext

Reply:
Done